crowbar_framework/app/models/service_object.rb

```diff
@@ -1190,6 +1193,15 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            databag["deployment"][@bc_name]["elements"]["to_remove"] ||= {}
+
+            old_nodes.each do |old_node|
+              @logger.debug "saving #{elem_remove} intention for #{old_node}"
+              databag["deployment"][@bc_name]["elements"]["to_remove"][old_node] ||= []
+              databag["deployment"][@bc_name]["elements"]["to_remove"][old_node] << elem_remove
```

Comment:
@aspiers I had to revert some of the changes that reduce duplication like 
* `to_remove = databag["deployment"][@bc_name]["elements"]["to_remove"]`
* `(to_remove[old_node] ||= []) << elem_remove`

the reference to the databag object gets lost (to_remove doesn't have it anymore), and subsequently things don't work anymore. That's why I reverted it